skimage/transform/radon_transform.py

```diff
@@ -63,25 +65,17 @@ def radon(image, theta=None, circle=True):
         theta = np.arange(180)
 
     if circle:
-        radius = min(image.shape) // 2
-        c0, c1 = np.ogrid[0:image.shape[0], 0:image.shape[1]]
-        reconstruction_circle = ((c0 - image.shape[0] // 2) ** 2
-                                 + (c1 - image.shape[1] // 2) ** 2)
-        reconstruction_circle = reconstruction_circle <= radius ** 2
-        if not np.all(reconstruction_circle | (image == 0)):
+        shape_min = min(image.shape)
+        radius = shape_min // 2
+        img_shape = np.array(image.shape)
+        coords = np.array(np.ogrid[:image.shape[0], :image.shape[1]])
+        dist = ((coords - img_shape // 2) ** 2).sum(0)
+        outside_reconstruction_circle = dist > radius ** 2
+        if np.any(image[outside_reconstruction_circle]):
             warn('Radon transform: image must be zero outside the '
                  'reconstruction circle')
         # Crop image to make it square
-        slices = []
-        for d in (0, 1):
-            if image.shape[d] > min(image.shape):
-                excess = image.shape[d] - min(image.shape)
-                slices.append(slice(int(np.ceil(excess / 2)),
-                                    int(np.ceil(excess / 2)
-                                        + min(image.shape))))
-            else:
-                slices.append(slice(None))
-        slices = tuple(slices)
+        slices = find_objects((~outside_reconstruction_circle).astype(int))[0]
```

Comment:
calling `find_objects` is more elegant, but it takes a few ms for a 2000x2000 image. This is not a big deal since the whole function takes several seconds to complete, but if someday we accelerate massively this function (the loop on angles could easily be parallelized) it could be significant. Even then it would only represent around 10% of the execution time so I'm ok with this change, I just wanted to raise this point in case somebody wants to comment.

Reply:
Seems silly to analyze the data instead of the metadata when looking for a slice....

Reply:
I agree with you, the use of `find_object` requires also negating `outside_reconstruction_circle` and casing it to `int`, which is a lot of processing to find the desired slice...
My last commit removes the call to `find_object`.